### PR TITLE
Anne-Sophie DEMESY - TODO list

### PR DESCRIPTION
**Pull request by asdemesy**

### CODE REVIEW

File: resources/fixtures_asdemesy.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
+INSERT INTO users (username, password, salt, role) VALUES
+('asdemesy', 'Q8U5bW+ruNRFvXdNd5xmiLpcSmZ9JS1FWPTPm6K/KbYNrwwk6xhvSL0AKkMBeg9HfDjBM2qNeBPTYk5cY/QoLg==', '4f66099d55095fa2c5ef3b3736cbd32749f233e0', 'ROLE_USER'),
+('alayacare', 'kEa36t7dY049btoyIpnXS4cHToS1ai/VtPBHpkxF+pZfvFDUf9cOmlT9NkDKsI6JVURhdrzELO89feDABjAQXA==', '4f66099d55095fa2c5ef3b3736cbd32749f233e0', 'ROLE_USER');
```
Comment (T0RAT0RA) [inline resources/fixtures_asdemesy.sql]: How would you migrate existing user?

File: src/App/Controller/ApiTodoController.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    /**
+     * Get Todo
+     * @return \Symfony\Component\HttpFoundation\JsonResponse
+     */
+    public function viewAction()
+    {
+        $id = $this->request->attributes->get('id');
+
+        if (!$id) {
+            $this->notFound();
```
Comment (T0RAT0RA) [inline src/App/Controller/ApiTodoController.php]: `notFound` doesn't seem to be implemented.

File: src/App/Controller/ApiTodoController.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        if ($this->cannotUpdate($todo->getUser()->getId())) {
+            // We don't return a 403, this way when an unauthorized user tries to access
+            // a resource, they don't know its existence
+            return $this->app->json('Not found', 404);
+        }
+
+        $data = array(
+            'id' => $todo->getId(),
+            'user_id' => $todo->getUser()->getId(),
+            'description' => $todo->getDescription(),
```
Comment (T0RAT0RA) [inline src/App/Controller/ApiTodoController.php]: Why not returning the status `completed` as well?

File: src/App/Repository/UserRepository.php [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+     *
+     * @param string $username The username
+     *
+     * @return UserInterface
+     *
+     * @throws UsernameNotFoundException if the user is not found
+     */
+    public function loadUserByUsername($username)
+    {
+       $queryBuilder = $this->db->createQueryBuilder();
```
Comment (T0RAT0RA) [inline src/App/Repository/UserRepository.php]: Why not using the `entity manager` here?
You wouldn't need `arrayToUser` to hydrate your object.

File: src/App/Entity/Todo.php [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+     *
+     * @ORM\Column(name="completed", type="boolean", nullable=false)
+     */
+    protected $completed = false;
+
+    /**
+     * Constraints
+     * @param ClassMetadata $metadata
+     */
+    static function loadValidatorMetadata(ClassMetadata $metadata) {
```
Comment (T0RAT0RA) [inline src/App/Entity/Todo.php]: Missing method visibility.

File: src/App/Controller/ApiTodoController.php (hunk 3 above)
Comment (asdemesy) [reply src/App/Controller/ApiTodoController.php]: Yes indeed, I did task 3 before task 2 and I forgot to update this controller

File: src/App/Controller/ApiTodoController.php (hunk 2 above)
Comment (asdemesy) [reply src/App/Controller/ApiTodoController.php]: Yes, I used to have this shortcut function `$this->app->json('Not found', 404);` inside `BaseController` but I forgot to remove this reference when I cleaned my code before committing

File: resources/fixtures_asdemesy.sql (hunk 1 above)
Comment (asdemesy) [reply resources/fixtures_asdemesy.sql]: I would use doctrine migration http://docs.doctrine-project.org/projects/doctrine-migrations/en/latest/reference/generating_migrations.html

File: src/App/Entity/Todo.php (hunk 5 above)
Comment (asdemesy) [reply src/App/Entity/Todo.php]: I'm not entirely sure what you mean ?

File: src/App/Repository/UserRepository.php (hunk 4 above)
Comment (asdemesy) [reply src/App/Repository/UserRepository.php]: I tried with em but I could not make it work properly before the dead line, which is why I left a comment on line 14. What I implemented is indeed not coherent with the rest of the code. em would have been better.

File: resources/fixtures_asdemesy.sql (hunk 1 above)
Comment (T0RAT0RA) [reply resources/fixtures_asdemesy.sql]: Yes, but how would you encode their passwords?

File: src/App/Controller/ApiTodoController.php (hunk 3 above)
Comment (T0RAT0RA) [reply src/App/Controller/ApiTodoController.php]: ok!

File: src/App/Controller/ApiTodoController.php (hunk 2 above)
Comment (T0RAT0RA) [reply src/App/Controller/ApiTodoController.php]: :+1:

File: src/App/Entity/Todo.php (hunk 5 above)
Comment (T0RAT0RA) [reply src/App/Entity/Todo.php]: `public`

Comment (asdemesy) [reply src/App/Entity/Todo.php]: yes of course !

File: resources/fixtures_asdemesy.sql (hunk 1 above)
Comment (asdemesy) [reply resources/fixtures_asdemesy.sql]: What I used for the script is the following encoder:
```
use Symfony\Component\Security\Core\Encoder\MessageDigestPasswordEncoder;
$encoder = new MessageDigestPasswordEncoder();
$password = $encoder->encodePassword('user1', 'alayacare');
```
I can also use other encode as specified in the doc (like BCrypt)

**Pull request closed without merging**